orttraining/orttraining/core/framework/distributed_run_context.h

```diff
@@ -10,15 +10,18 @@
 namespace onnxruntime {
 namespace training {
 enum WorkerGroupType {
-  DataParallel = 0,
-  HorizontalParallel = 1,
-  ModelParallel = 2,
-  WorkerGroupTypeCount = 3,
+  GlobalParallel = 0,
+  DataParallel = 1,
+  NodeLocalDataParallel = 2,
+  CrossNodeDataParallel = 3,
```

Comment:
Suggest adding some comments here on the different groups, and relationship between them